Loop/Managers/CGMStalenessMonitor.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  CGMStalenessMonitor.swift
+//  Loop
+//
+//  Created by Pete Schwamb on 10/15/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+import LoopKit
+import LoopCore
+
+protocol CGMStalenessMonitorDelegate: class {
+    func getLatestCGMGlucose(since: Date, completion: @escaping (_ result: Swift.Result<StoredGlucoseSample?, Error>) -> Void)
+}
+
+class CGMStalenessMonitor {
+    
+    private let log = DiagnosticLog(category: "CGMStalenessMonitor")
+    
+    private var cgmStalenessTimer: Timer?
+    
+    weak var delegate: CGMStalenessMonitorDelegate? = nil {
+        didSet {
+            if delegate != nil {
+                checkCGMStaleness()
+            }
+        }
+    }
+
+    @Published var cgmDataIsStale: Bool = true {
+        didSet {
+            self.log.debug("cgmDataIsStale: %{public}@", String(describing: cgmDataIsStale))
+        }
+    }
+
+    private static var cgmStalenessTimerTolerance: TimeInterval = .seconds(10)
+    
+    public func cgmGlucoseSamplesAvailable(_ samples: [NewGlucoseSample]) {
+        let mostRecentGlucose = samples.map { $0.date }.max()!
```

Comment:
[nit] Consider precondition that `samples` is not empty since this is a public func. Or a guard statement.

Reply:
Feels a little overly defensive (the name says "samples are available"), but I can add a guard.